src/mscorlib/src/System/Array.cs

```diff
@@ -680,17 +680,27 @@ Int32 IStructuralComparable.CompareTo(Object other, IComparer comparer)
                 ThrowHelper.ThrowArgumentException(ExceptionResource.ArgumentException_OtherNotArrayOfCorrectLength, ExceptionArgument.other);
             }
 
+            if (o.Length == 0)
+            {
+                return 0;
+            }
+
+            if (comparer == null)
+            {
+                comparer = Comparer.Default;
```

Comment:
Not sure on this, but I went with `Comparer.Default` just in Array to be consistent with everything else in array uses it. However, I used `Comparer<object>.Default` in Tuple and ValueTuple to be consistent with everything else in the BCL.
/cc @jcouv